### DISCUSSION TITLE
How to keep page alive?

### DISCUSSION THREAD

**Discussion by toniengelhardt** [nuxt/nuxt#16572] [Questions]
I have a page `/journal` with an infinite list and complicated state that I want to keep alive when I navigate to `/journal/[id]` and back. I tried both setting `keepalive: true` in the `Journal.vue` page `definePageMeta` and wrapping the list in a Vue `<KeepAlive>` tag, but both won't work and `Journal.vue` `onMounted` hook is always fired when navigating back.

_Note: `/journal/[id]` is not a nested route of `/journal`_.

Is there any way to do this?

**Comment by IsraelOrtuno** [comment]
I was able to make this work using this in `:keepalive: {}` in `<NuxtPage>` tag in the `app.vue`:

```vue
<NuxtPage :keepalive="{}" />
```

Was not able to make it work with `definePageMeta` for now nor specifying any include.

**Reply by IsraelOrtuno** [reply]
A way I could make this work was going the other way round. Setting `app: { keepalive: true}` in my `nuxt.config.ts` and then disabling it in any page I do not want it with:

```js
definePageMeta({ keepalive: false })
```

**Comment by IsraelOrtuno** [comment]
It's more a definition you should set in your page to keep its children alive. I was confused about the behaviour as well. Refer to @danielroe comment 👇

https://github.com/nuxt/framework/issues/8367#issuecomment-1286628219

**Reply by toniengelhardt** [reply]
Thanks @IsraelOrtuno, but I still can't make it work, see example in https://github.com/nuxt/framework/issues/8367#issuecomment-1286628219.

**Reply by toniengelhardt** [reply]
The idea, based on your explanation, is basically to wrap the pages that should be kept alive in a common parent. But no luck unfortunately...
